### ISSUE TITLE
nativeSetEnv from p4a Qt bootstrap not working

### ISSUE THREAD

**Issue by accumulator** [spesmilo/electrum#8119]
The bootstrap C entry point needs some environment variables to initialize the python interpreter, which are set from the Java activity code. However, nativeSetenv calls seem to be ineffective.

`[p4a-bootstrap-qt]/.../PythonActivity.nativeSetenv` is declared as extern, and implemented in `[p4a-bootstrap-common]/.../start.c`

As a workaround, all environment variables are not set in the environment, but instead added to `PythonActivity/QtActivity`  as an `extraenvvars` attribute containing a base64 encoded string of kv pairs, which is internally put in the environment by `QtActivity`. This works _only_ for debug builds, not for release builds.

**Comment by accumulator** [comment]
fixed in 6a1f19fd86d7869fdbea952b8cf84fa5592bbb41

**Comment by accumulator** [comment]
It turns out this was hard to pin down because the debugger has a different view of the available symbols in `libmain.so` the moment we were calling `nativeSetenv`. The JNI symbols only became available to the Java VM after `JNI_OnLoad` was called by loading of the library by Qt5, but this was _after_ we already tried to call `nativeSetenv`. The debugger naively assumed we ended up in `Java_org_kivy_android_PythonActivity_nativeSetenv` while in reality the Java VM silently ignored the call.
